mw-webapp/src/utils/DateUtils.ts

```diff
@@ -48,21 +49,22 @@ export class DateUtils {
    * Get dates between two dates
    */
   public static getDatesBetween(startDate: Date, endDate: Date): Date[] {
-    const timeDifferenceInMilliseconds = Math.abs(endDate.getTime() - startDate.getTime());
-    const daysAmountInRange = Math.ceil(timeDifferenceInMilliseconds / DAY_MILLISECONDS);
-    const datesList = new Array(daysAmountInRange)
-      .fill(null)
-      .map((item, i) => {
 
-        /**
-         * TODO #573: Correction need because method getShortISODateValue return time in GMT format but we need UTC format
-         */
-        const correction = DAY_MILLISECONDS;
+    const isoStartDate = DateUtils.getShortISODateValue(startDate);
+    const isoEndDate = DateUtils.getShortISODateValue(endDate);
 
-        return new Date(startDate.getTime() + (i * DAY_MILLISECONDS) + correction);
-      });
+    const currentDate = new Date(isoStartDate);
+    const lastDate = new Date(isoEndDate);
```

Comment:
Can you please explain why do you need convert Date to ISO and then again convert to Date/. It's looking strange

Reply:
![image](https://github.com/tritonJS826/masters-way/assets/112805668/6d8979f2-6022-4039-bc27-b193de294133)

This is necessary because the display is in one format(ISO), while calculations are in a different format, causing data discrepancies. However, if the initial value(currentDate ) is fixed, the display will be correct.

Reply:
the end date is fixed in the same way

Reply:
since in the chart we only need whole dates (without minutes and seconds), when we have a fixed start and end date, we can easily calculate the number of days between dates
```
const daysCount = (lastDate.getTime() - currentDate.getTime()) / DAY_MILLISECONDS;
```